### PR TITLE
NEW (re)generate PDF for notification emails (if not existing)

### PR DESCRIPTION
**Pull request by ptibogxiv**
new PR of  #16900 
with reload object for correct (refrence) directory for pdf storage and correct ref with substitution in templated mail

### CODE REVIEW

File: htdocs/core/class/notify.class.php [hunk 1]
```diff
@@ -577,30 +577,58 @@ public function send($notifcode, $object, $filename_list = array(), $mimetype_li
...
+								$newlang = '';
+
+								if ($conf->global->MAIN_MULTILANGS && empty($newlang)) {
+										  $newlang = $object->thirdparty->default_lang;
+								}
+								if (!empty($newlang)) {
+									$outputlangs = new Translate("", $conf);
+									$outputlangs->setDefaultLang($newlang);
+								}
+								$modelpdf = !empty($object->modelpdf)?$object->modelpdf:$conf->global->FACTURE_ADDON_PDF;
```
Comment (eldy) [inline htdocs/core/class/notify.class.php:591]: Doc will not always be an invoice here. So use of $conf->global->FACTURE_ADDON_PDF must be done only if object->element is 'facture'
You must also deal with the case modelpdf is empty. You can just exclude generation of doc in such a case.

Comment (ptibogxiv) [reply htdocs/core/class/notify.class.php]: fix done

File: htdocs/core/class/notify.class.php [hunk 2]
```diff
@@ -577,30 +588,58 @@ public function send($notifcode, $object, $filename_list = array(), $mimetype_li
 						$pdf_path = $dir_output."/".$ref.".pdf";
 						if (!dol_is_file($pdf_path)) {
 							// We can't add PDF as it is not generated yet.
-							$filepdf = '';
-						} else {
+							if (empty($conf->global->MAIN_DISABLE_PDF_AUTOUPDATE)) {
```
Comment (eldy) [inline htdocs/core/class/notify.class.php:591]: I would add a test on method_exists($object, 'generateDocument')  too here.

Comment (ptibogxiv) [reply htdocs/core/class/notify.class.php]: done !

**Pull request closed without merging**